### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by tpravin**

### CODE REVIEW

File: app/routes.php [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        list($header, $payload, $signature) = explode($separator, $jwt_access_token);
+        return json_encode(json_decode(base64_decode($payload)), JSON_PRETTY_PRINT);
+    }
+
+    /**
+     * @param string|null $wellKnownUrl
+     * @param string|null $element
+     * @return string|null
+     */
+    function get_location_from_metadata(?string $wellKnownUrl, string $element): string
```
Comment (tshearouse) [inline app/routes.php]: You have a mix of camelCase and snake_case function names in this class. Please standardize on one naming convention across the entire project.

File: src/Domain/User/UserRepository.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+     * @return User[]
+     */
+    public function findAll(): array;
+
+    /**
+     * @param int $id
+     * @return User
+     * @throws UserNotFoundException
+     */
+    public function findUserOfId(int $id): User;
```
Comment (tshearouse) [inline src/Domain/User/UserRepository.php]: More typical naming convention would be `findUserById`, as opposed to `findUserOfId`.

File: src/Infrastructure/Persistence/User/InMemoryUserRepository.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+     * @param array|null $users
+     */
+    public function __construct(array $users = null)
+    {
+        $this->users = $users ?? [
+            1 => new User(1, 'bill.gates', 'Bill', 'Gates'),
+            2 => new User(2, 'steve.jobs', 'Steve', 'Jobs'),
+            3 => new User(3, 'mark.zuckerberg', 'Mark', 'Zuckerberg'),
+            4 => new User(4, 'evan.spiegel', 'Evan', 'Spiegel'),
+            5 => new User(5, 'jack.dorsey', 'Jack', 'Dorsey'),
```
Comment (tshearouse) [inline src/Infrastructure/Persistence/User/InMemoryUserRepository.php]: Why are these needed? Actually, why are users needed at all - is the intent to contrast between users of a connected software product vs MJD users? If that is valuable, can we call it out more explicitly?

**Comment by jconner0623** [comment]
@tshearouse Pravin has moved on from API Support. Are you ok with me merging this for now, and we can iterate later as we have time or with feedback? Or are your comments blockers to merge?
